### PR TITLE
feat: integrate block number loading for date-to-block conversion

### LINKED ISSUE #178: Feat: Integrate block number loading for date-to-block conversion
## Context

The scanner has date ranges but needs block numbers to query the blockchain. Block numbers are already available from the BlockFinder component and must be integrated into the scanning flow.

## Current Behavior

1. Scanner determines date ranges to process
2. Cannot convert dates to block numbers for RPC queries
3. Block data exists but isn't being used

## Expected Behavior

For each date in the range to be scanned:
- Load block numbers using `fileManager.readBlockNumbers(chainId)`
- Convert start/end dates to block numbers
- Handle missing block number data by throwing descriptive error
- Create block ranges for each date (start of day to end of day)
- Validate all required block numbers are available before proceeding

## Required Context

- `src/recipient-recieved-scanner.ts` - current implementation
- `src/balance-fetcher.ts` - pattern for block number usage
- `src/types.ts` - BlockNumbers structure
- Block numbers are keyed by date in YYYY-MM-DD format

## Out of Scope

- Chunking block ranges
- Querying events from blockchain
- Parsing or storing events
- Retry logic for RPC calls

### PR DESCRIPTION
**Pull request by yahgw-ai**
## Summary

Implements block number loading and date-to-block conversion in RecipientRecievedScanner as specified in issue #178.

## Changes

- Added `convertDatesToBlockRanges()` method to convert date ranges to block number ranges
- Added `validateBlockNumbersForDates()` to ensure all required block numbers exist before processing
- Added `convertDateToBlockRange()` to convert individual dates to start/end block pairs
- Implemented proper calculation of start blocks (previous day's end block + 1, or block 1 for first date)
- Added comprehensive error handling with descriptive messages for missing block numbers

## Test Plan

Followed strict TDD approach:
- RED phase: Added comprehensive failing tests for all requirements
- GREEN phase: Implemented minimal code to make tests pass
- REFACTOR phase: Cleaned up implementation for clarity and maintainability

All new tests pass, validating:
- ✅ Date to block number conversion
- ✅ Error handling for missing block numbers
- ✅ Block range creation with start/end blocks
- ✅ Validation before processing
- ✅ Edge case handling (first date in block data)

## Notes

- This PR only handles the conversion logic as specified in the issue
- Actual event querying, chunking, and processing are out of scope (marked with TODO)
- Implementation is ready for the next phase of development

Closes #178

### CODE REVIEW

File: src/recipient-recieved-scanner.ts [hunk 1]
```diff
@@ -81,7 +81,15 @@ export class RecipientRecievedScanner {
         continue;
       }
 
-      // TODO: Process the date range (out of scope for this issue)
+      // Convert dates to block ranges and validate block numbers
+      const blockRanges = this.convertDatesToBlockRanges(
```
Comment (yahgwai) [inline src/recipient-recieved-scanner.ts]: We're overcomplicating this. We don't need to first calculate date ranges and then calculate block ranges from that. We can just directly calculate block ranges.

**Review by yahgwai — DISMISSED** [review]

**Review by yahgwai — COMMENTED** [review]
reoipening comment - unnapprove

**Review by yahgwai — APPROVED** [review]
